File: classifier/data/raw/ai/samples/project_euler_java/Problem077.java
// Project Euler 77: first integer expressible as a sum of primes in over 5000 ways.
import java.util.ArrayList;
import java.util.List;

public class Problem077 {
    public static void main(String[] args) {
        for (int n = 2; ; n++) {
            long[] ways = new long[n + 1];
            ways[0] = 1;
            for (int prime : primesUpTo(n)) {
                for (int sum = prime; sum <= n; sum++) {
                    ways[sum] += ways[sum - prime];
                }
            }
            if (ways[n] > 5000) {
                System.out.println(n);
                return;
            }
        }
    }

    private static List<Integer> primesUpTo(int limit) {
        boolean[] composite = new boolean[limit + 1];
        List<Integer> primes = new ArrayList<>();
        for (int n = 2; n <= limit; n++) {
            if (!composite[n]) {
                primes.add(n);
                if ((long) n * n <= limit) {
                    for (int multiple = n * n; multiple <= limit; multiple += n) {
                        composite[multiple] = true;
                    }
                }
            }
        }
        return primes;
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_077.zig
// Project Euler Problem 77: first integer expressible as a sum of primes in over 5000 ways.
const std = @import("std");

pub fn main() !void {
    const max = 1_000;
    var composite = [_]bool{false} ** (max + 1);
    var p: usize = 2;
    while (p * p <= max) : (p += 1) {
        if (!composite[p]) {
            var multiple = p * p;
            while (multiple <= max) : (multiple += p) composite[multiple] = true;
        }
    }

    var answer: usize = 0;
    var target: usize = 2;
    while (target <= max) : (target += 1) {
        var ways = [_]u32{0} ** (max + 1);
        ways[0] = 1;
        var prime: usize = 2;
        while (prime <= target) : (prime += 1) {
            if (composite[prime]) continue;
            var sum = prime;
            while (sum <= target) : (sum += 1) ways[sum] += ways[sum - prime];
        }
        if (ways[target] > 5_000) {
            answer = target;
            break;
        }
    }

    var buffer: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(std.Io.Threaded.global_single_threaded.io(), &buffer);
    try out.interface.print("{d}\n", .{answer});
    try out.interface.flush();
}
